src/main/kotlin/org/cascadebot/cascadebot/utils/lists/WeightedList.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.utils.lists
+
+import org.cascadebot.cascadebot.utils.WeightPair
+import java.util.Random
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+
+class WeightedList<T : Any> {
+
+    companion object {
+        @Transient
+        val random: Random = Random()
```

Comment:
A nice option, but not strictly necessary it to allow a custom random seed.

Reply:
That seems very specific, I *can* add it

Reply:
> A nice option, but not strictly necessary

open for discussion. I just thought it would be nice if we wanted a custom seed for future stuff (or even let users set a seed on welcomes)

Reply:
I can add it since it's a relatively small thing. If we don't use it, it can be removed